lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -258,6 +259,23 @@ def configure_app_extension_api_only_for_target(target)
             config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'YES'
           end
         end
+
+        def user_swift_version
+          declared_swift_versions = analysis_result
+            .targets
+            .map(&:user_project)
+            .compact
+            .flat_map(&:native_targets)
+            .flat_map(&:build_configurations)
+            .flat_map(&:build_settings)
+            .map { |settings| settings['SWIFT_VERSION'] }
+            .compact
+            .uniq
+
+          # TODO: Assert only 1 or 0 unique Swift Versions declared
+
+          declared_swift_versions.first
```

Comment:
If I had a user project with 3 targets, two on `2.3` and 1 on `3.0` won't this always pick up the first in the set? As it goes back to the global list in the user project, and then makes a list of them all, and all their targets. 

Perhaps a `select` after `.flat_map(&:native_targets)` to scope it to only the initially named targets? 
This may also have to assert if they have different swift values too alas